platform/src/main/java/org/stellar/anchor/platform/configurator/ConfigMap.java

```diff
@@ -70,7 +72,16 @@ public Map<String, String> toStringMap() {
   }
 
   public void merge(ConfigMap config) {
+    // TODO: instead of merging, create a property source with higher precedence
+    // Matches any string of the form: <listName>[<index>].<elementName>
```

Comment:
How does it work with .yaml files?

Reply:
YAML will produce a config map such that the keys of the map are the flattened property names. 

For example, this will create the key `platform.server`:
```
platform:
  server: 8085
```

Reply:
I'm more curious about arrays, will it also create `client[0].name` property?

Reply:
It does.

```yaml
clients:
  - name: referenceCustodial
    type: custodial
    signing_key: GDJLBYYKMCXNVVNABOE66NYXQGIA5AC5D223Z2KF6ZEYK4UBCA7FKLTG
    callback_url: http://wallet-server:8092/callbacks
```
will create 
```json
        "clients[0].name": {
            "value": "referenceCustodial",
            "source": "DEFAULT"
        },
...
```